contracts/lpp/src/lpp.rs

```diff
@@ -78,15 +76,41 @@ where
         Ok(LiquidityPool { config, total })
     }
 
-    pub fn total_lpn(&self, deps: &Deps<'_>, env: &Env) -> Result<Coin<LPN>> {
-        let res = self.balance(&env.contract.address, &deps.querier)?
-            + self.total.total_principal_due()
-            + self.total.total_interest_due_by_now(env.block.time);
+    pub fn deposit_capacity(
+        &self,
+        querier: &QuerierWrapper<'_>,
+        env: &Env,
+        pending_deposit: Option<Coin<Lpn>>,
```

Comment:
Why do we complicate the logic? Do not think we need this argument ... we shall calculate how much deposit amount this Lpp can accept.

Reply:
By the time this code gets executed, the funds have already arrived. That is why I was checking whether it is **above** the limit **when they have arrived**, the previous implementation.